python/py-jsonschema/Portfile

```diff
@@ -65,6 +66,18 @@ if {${name} ne ${subport}} {
             depends_lib-append  port:py${python.version}-jsonschema-specifications \
                                 port:py${python.version}-referencing \
                                 port:py${python.version}-rpds-py
+
+            variant format_nongpl description {Build the 'format-nongpl' extra needed for Jupyter 4} {
+                depends_lib-append  port:py${python.version}-fqdn \
+                                    port:py${python.version}-idna \
+                                    port:py${python.version}-isoduration \
+                                    port:py${python.version}-jsonpointer \
+                                    port:py${python.version}-rfc3339-validator \
+                                    port:py${python.version}-rfc3986-validator \
+                                    port:py${python.version}-uri-template \
+                                    port:py${python.version}-webcolors
+            }
+            default_variants    +format_nongpl
```

Comment:
setting (or changing) a default variant requires to increase the `revision`

Reply:
It's listed as an "extra" in the pypi package. No point in installing extra files unless you need them. So having the option of installing less clutter is good.

Reply:
@reneeotten I doubt @stromnov is ever going to comment.

Reply:
Firstly, I think we could drop EOL support (pre-3.8) of Python.

Secondly, I think it's not a good idea to force the +format_nongpl variant only to implicitly support the Jupyter ecosystem. This can be done in alternative ways: by explicitly writing dependencies in Jupiter or by creating a meta port (for example, py-jsonschema-jupyter).

Reply:
Okay, @johankytt and @barracuda156, let's drop 3.7 etc., and please otherwise follow @stromnov's requests.

Reply:
How do you create a meta port that installs the jsonschema variant if direct variant dependency is not possible? Or do you mean the complete duplication of the jsonschema+nongpl port? I think I've seen some kind of a meta port somewhere before, but can't find an example at the moment.